invoices/tenkfeet_api.py

```diff
@@ -2,205 +2,222 @@
+        "updated_at": opt(Use(parse_datetime)),
         "use_parent_bill_rates": bool
     }])
 
-    HOUR_ENTRIES_SCHEMA = schema.Schema([
-        [
-            int,
-            str,
-            int,
-            str,
-            str,
-            str,
-            str,
-            str,  # Date (Friday, 29.12.2017), could be parsed
-            float,
-            float,
-            float,
-            float,
-            float,
-            float,
-            str,
-            schema.Or(str, NONETYPE),
-            str,
-            str,
-            str,
-            str,
-            str,
-            int,
-            str,
-            schema.Or(str, NONETYPE),
-            schema.Or(str, NONETYPE),
-            str,  # Date (2017-01-01)
-            str,  # Date (2017-01-01)
-            schema.Or(str, NONETYPE),
-            float,
-            str,  # email
-            int,
-            str,
-            schema.Or(str, NONETYPE),
-            str,
-            schema.Or(str, NONETYPE),
-            str,
-            str,
-            int,
-            int,
-            int,
-            str,  # Date (2017-01-01)
-            str,
-            str,
-            str,
-            int,
-            int,
-            int,
-            str,
-            float,
-            str,  # YYYY-MM
-            str,
-            int,
-            str,
-            schema.Or(str, NONETYPE),
-            schema.Or(str, NONETYPE),
-            str,
-            str,
-            str,
-            str,
-            str,
-        ]
-    ])
+    TIME_ENTRIES_SCHEMA = Schema([{
+        "id": int,
+        "assignable_id": int,
+        "assignable_type": str,
+        "user_id": int,
+        "bill_rate": Use(parse_float),
+        "bill_rate_id": opt(int),
+        "date": Use(parse_date),
+        "hours": Use(parse_float),
+        "scheduled_hours": Use(parse_float),
+        "notes": opt(str),
+        "task": opt(str),
+        "is_suggestion": bool,
+        "created_at": Use(parse_datetime),
+        "updated_at": Use(parse_datetime),
+        "approvals": {
+            "data": [{
+                "id": int,
+                "status": str,
+                "approvable_id": int,
+                "approvable_type": str,
+                "submitted_by": int,
+                "submitted_at": Use(parse_datetime),
+                "approved_by": opt(int),
+                "approved_at": opt(Use(parse_datetime)),
+                "created_at": Use(parse_datetime),
+                "updated_at": Use(parse_datetime)
+            }],
+            "paging": {
+                "next": opt(str),
+                "page": int,
+                "per_page": int,
+                "previous": opt(str),
+                "self": str,
+            }
+        }
+    }])
 
     def __init__(self, apikey):
         self.apikey = apikey
         self.logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
+    def api_url(self, path):
```

Comment:
FYI: instead of this `if` thing, see `params` in `requests`.

Reply:
Yeah just yesterday used `params` in my own project, should probably refactor code more.